### DISCUSSION TITLE
Additional "is" utils

### DISCUSSION THREAD

**Discussion by ababkov** [rexlabsio/enum-php#12] [Q&A] [answered]
# Overview

Thoughts on adding the following utilities to augment the current is case?

`isNot($enum)` - True if enum is not a value
`isAnyOf($enum[])`
`isNoneOf($enum[])`

I find myself needing these often and instead end up having to revert to inArray checks on the key + statics which i don't love.

# Examples

## isNot

Previous:

* `$invoice->status->is(InvoiceStatus::AWAITING_PAYMENT()) === false`
* `!$invoice->status->is(InvoiceStatus::AWAITING_PAYMENT())`

New:

* `$invoice->status->isNot(InvoiceStatus::AWAITING_PAYMENT())`

## isNoneOf

Previous:

```
!in_array(
  $invoice->status->key(),
  [
    InvoiceStatus::DRAFT,
    InvoiceStatus::AWAITING_PAYMENT,
  ],
  true
)
```

```
$invoice->status->is(InvoiceStatus::DRAFT()) === false &&
  $invoice->status->is(InvoiceStatus::AWAITING_PAYMENT()) === false
```

New:

```
$invoice->status->isNoneOf([
  InvoiceStatus::DRAFT(),
  InvoiceStatus::AWAITING_PAYMENT(),
])
```

**Comment by colemanator** [comment]
I like it, makes sense and is probably easier to read

**Comment by lachlankrautz** [comment]
Yeah those seem pretty readable.

**Comment by ababkov** [comment, marked as answer]
Pull Request https://github.com/rexlabsio/enum-php/pull/13
